internal/cfgmgr/cfgmgr.go

```diff
@@ -0,0 +1,150 @@
+		}
+		propertyType := uint32(0)
+		propertyBufferSize := uint32(0)
+
+		// get the size of the property buffer by querying with a nil buffer and zeroed propertyBufferSize
+		err = cmGetDevNodeProperty(devNodeInst, devPKeyDeviceLocationPaths, &propertyType, nil, &propertyBufferSize, 0)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get property buffer size of devnode query for %s with", id)
+		}
+
+		// get the property with the resulting propertyBufferSize
+		propertyBuffer := make([]uint16, propertyBufferSize/2)
+		err = cmGetDevNodeProperty(devNodeInst, devPKeyDeviceLocationPaths, &propertyType, &propertyBuffer[0], &propertyBufferSize, 0)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get location path property from device node for %s with", id)
+		}
+		if propertyType != _DEVPROP_TYPE_STRING_LIST {
+			return nil, fmt.Errorf("expected to return property type DEVPROP_TYPE_STRING_LIST %d, instead got %d", _DEVPROP_TYPE_STRING_LIST, propertyType)
+		}
+		if int(propertyBufferSize/2) > len(propertyBuffer) {
+			return nil, fmt.Errorf("location path is too large for the buffer, size in bytes %d", propertyBufferSize)
+		}
+
+		formattedResult, err := convertFirstNullTerminatedValueToString(propertyBuffer[:propertyBufferSize/2])
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, formattedResult)
+	}
+
+	return result, nil
+}
+
+// helper function that finds the first \u0000 rune and returns the wide string as a regular go string
+func convertFirstNullTerminatedValueToString(buf []uint16) (string, error) {
+	r := utf16.Decode(buf)
+	converted := string(r)
+	zerosIndex := strings.IndexRune(converted, '\u0000')
+	if zerosIndex == -1 {
+		return "", errors.New("cannot convert value to string, malformed data passed")
+	}
+	return converted[:zerosIndex], nil
+}
+
+func GetChildrenFromInstanceIDs(parentIDs []string) ([]string, error) {
+	var result []string
+	for _, id := range parentIDs {
+		pszFilterParentID := []byte(id)
+		children, err := getDeviceIDList(&pszFilterParentID[0], _CM_GETIDLIST_FILTER_BUSRELATIONS)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, children...)
+	}
+	return result, nil
+}
+
+func getDeviceIDList(pszFilter *byte, ulFlags uint32) ([]string, error) {
+	listLength := uint32(0)
+	if err := cmGetDeviceIDListSize(&listLength, pszFilter, ulFlags); err != nil {
+		return nil, err
+	}
+	if listLength == 0 {
+		return []string{}, nil
+	}
+	buf := make([]byte, listLength)
+	if err := cmGetDeviceIDList(pszFilter, &buf[0], uint32(listLength), ulFlags); err != nil {
+		return nil, err
+	}
+
+	return convertStringSetToSlice(buf)
+}
+
+// convertStringSetToSlice is a helper function used to convert the contents of
+// `buf` into a string slice. `buf` contains a set of null terminated strings
+// with an additional null at the end to indicate the end of the set.
+func convertStringSetToSlice(buf []byte) ([]string, error) {
```

Comment:
I feel it might be a good idea to put `convertFirstNullTerminatedValueToString` and `convertStringSetToSlice` functions in a separate directory named `util` or something. What do you think?

Reply:
I'm not sure moving `convertFirstNullTerminatedValueToString` is super useful since it has a very specific use case: specifically when you have a list of null terminated strings returned but you only want the first one. But I do think moving `convertStringSetToSlice` would be good :)

Reply:
I've moved `convertStringSetToSlice` let me know if that works!